### ISSUE TITLE
Using typesFile with ESM/moduleResolution nodenext causes build error due to missing ".js" extension

### ISSUE THREAD

**Issue by joshleitzel** [ardeois/graphql-codegen-typescript-mock-data#163]
When using the `config.typesFile` option in a TypeScript project with `moduleResolution` set to `node16` or `nodenext`, this package will always cause an error when building the project. This is because TypeScript expects local imports to have a `.js` extension. For example this config:

```typescript
config: {
  typesFile: "types/generated"
}
```

will generate an import like this:

```typescript
import { Type1, Type2 } from "types/generated";
```

which will in turn result in the error:

```bash
Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean 'types/generated.js'?
```

And you can't work around it by appending the `.js` yourself:

```typescript
config: {
  typesFile: "types/generated.js"
}
```

because the extension is [explicitly stripped](https://github.com/ardeois/graphql-codegen-typescript-mock-data/blob/e899d31c5123ffce4ff9539be97a2d95cb99ca25/src/index.ts#L800).

I'm not sure what the permanent solution here is, but I suspect the higher-level config's [`emitLegacyCommonJSImports`](https://the-guild.dev/graphql/codegen/docs/getting-started/esm-typescript-usage) value should be respected.

I worked around this for now by patch-packaging the source code to remove the extension stripping.

**Comment by ardeois** [comment]
Yes you're right we should probably add support for `emitLegacyCommonJSImports` option

Feel free to open a PR
